data_diff/dbt.py

```diff
@@ -162,12 +157,12 @@ def _get_diff_vars(
         prod_schema = dev_schema
 
     if dbt_parser.requires_upper:
-        dev_qualified_list = [x.upper() for x in [dev_database, dev_schema, model.alias]]
-        prod_qualified_list = [x.upper() for x in [prod_database, prod_schema, model.alias]]
+        dev_qualified_list = [x.upper() for x in [dev_database, dev_schema, model.alias] if x]
+        prod_qualified_list = [x.upper() for x in [prod_database, prod_schema, model.alias] if x]
         primary_keys = [x.upper() for x in primary_keys]
     else:
-        dev_qualified_list = [dev_database, dev_schema, model.alias]
-        prod_qualified_list = [prod_database, prod_schema, model.alias]
+        dev_qualified_list = list(filter(None, [dev_database, dev_schema, model.alias]))
+        prod_qualified_list = list(filter(None, [prod_database, prod_schema, model.alias]))
```

Comment:
Generally, list comprehensions are considered to be better than iterators, especially than iterators instantly converted to lists. So, to align with the style a few lines above:

```suggestion
        dev_qualified_list = [x for x in [dev_database, dev_schema, model.alias] if x]
        prod_qualified_list = [x for x in [prod_database, prod_schema, model.alias] if x]
```